pandas/core/arrays/datetimelike.py

```diff
@@ -710,7 +710,7 @@ def _from_factorized(cls, values, original):
         return cls(values, dtype=original.dtype)
 
     def _values_for_argsort(self):
-        return self._data
+        return self._data.view("M8[ns]")
```

Comment:
@jbrockmendel moved this down to the arrays. Basic idea is to just sort as datetime64[ns], which should be correct for datetime, timedelta, and period.

Reply:
sounds good, thanks